src/btree/bt_cursor.c

```diff
@@ -1124,25 +1123,21 @@ __wt_btcur_remove(WT_CURSOR_BTREE *cbt, bool positioned)
     WT_ERR(__wt_cursor_func_init(cbt, true));
 
     if (btree->type == BTREE_ROW) {
-        WT_ERR_NOTFOUND_OK(__cursor_row_search(cbt, false, NULL, NULL), true);
-        if (ret == WT_NOTFOUND)
-            goto search_notfound;
+        WT_ERR(__cursor_row_search(cbt, false, NULL, NULL));
 
         /* Check whether an update would conflict. */
         WT_ERR(__curfile_update_check(cbt));
 
         if (cbt->compare != 0)
-            goto search_notfound;
+            WT_ERR(WT_NOTFOUND);
         WT_WITH_UPDATE_VALUE_SKIP_BUF(ret = __wt_cursor_valid(cbt, cbt->tmp, WT_RECNO_OOB, &valid));
         WT_ERR(ret);
         if (!valid)
-            goto search_notfound;
+            WT_ERR(WT_NOTFOUND);
```

Comment:
This new behaviour needs to be documented. Earlier cursor->remove operation doesn't return any error. Also, the behaviour WT_CURSTD_OVERWRITE needs to be modified that cursor->remove.